legal-api/src/legal_api/resources/v2/business/business_filings/business_filings.py

```diff
@@ -620,12 +620,25 @@ def save_filing(client_request: LocalProxy,  # pylint: disable=too-many-return-s
                 datetime.datetime.fromisoformat(filing.filing_json['filing']['header']['effectiveDate']) \
                 if filing.filing_json['filing']['header'].get('effectiveDate', None) else datetime.datetime.utcnow()
 
+            filing.hide_in_ledger = ListFilingResource._hide_in_ledger(filing)
             filing.save()
         except BusinessException as err:
             return None, None, {'error': err.error}, err.status_code
 
         return business or bootstrap, filing, None, None
 
+    @staticmethod
+    def _hide_in_ledger(filing: Filing) -> bool:
+        """Hide the filing in the ledger."""
+        hide_in_ledger = False
+        if (filing.filing_type == 'adminFreeze' or
+            (filing.filing_type == 'dissolution' and filing.filing_sub_type == 'involuntary')):
+            hide_in_ledger = True
+        elif (jwt.validate_roles([SYSTEM_ROLE]) and 
+              str(request.headers.get('hide-in-ledger', None)).lower() == 'true'):
+            hide_in_ledger = True
```

Comment:
This will be used for putBackOff filing for an expired limited restoration filing.